pkg/hashreader/hashreader.go

```diff
@@ -0,0 +1,120 @@
+package hashreader
+
+// HashReader writes what it reads from an io.Reader to an MD5 and SHA256 hash.Hash.
+import (
+	"bytes"
+	"crypto/md5"
+	"crypto/sha256"
+	"encoding/hex"
+	"hash"
+	"io"
+)
+
+// HashReader verifies that the content of the io.Reader matches the expected checksums.
+type HashReader struct {
+	md5sum, sha256sum []byte // Byte values of md5sum, sha256sum of client sent values.
+
+	src                 io.Reader
+	size                int64
+	md5Hash, sha256Hash hash.Hash
```

Comment:
I think checksum and hasher need to be grouped i.e. md5sum and md5hash grouped together.  Also I personally like `md5Hasher` over `md5Hash`

Reply:
hasher is not go style here we just keep it similar to hash.Hash